src/VirtualDevice.ts

```diff
@@ -260,7 +260,7 @@ export class VirtualDevice {
                     if (response.statusCode === 200) {
                         const result = this.handleBatchResponse(data as string);
                         if ((result as IVirtualDeviceError).error) {
-                            reject(new Error((result as IVirtualDeviceError).error));
+                            reject(result);
```

Comment:
is this gonna change the object we get on the catch? because it is weird to receive a non Exception object on the try/catch block.

Reply:
yes, this is going to change the error. We were already doing this for the not async mode, but I agree that is not ideal. 
We can create another ticket to refactor both "Error" responses and filter out correctly in skill testing